### ISSUE TITLE
Race condition in docker.Client implementation

### ISSUE THREAD

**Issue by ghost** [ory/dockertest#290]
**Describe the bug**

The `Client` in the `docker` sub-package is not concurrency safe if you try to start multiple containers concurrently. 
I observed this while running tests with the race detector enabled `go test -race`. Below is the output (some lines have been redacted but are not relevant).

```
==================
WARNING: DATA RACE
Write at 0x00c0001622b8 by goroutine 37:
  github.com/ory/dockertest/docker.(*Client).checkAPIVersion()
      /home/runner/go/pkg/mod/github.com/ory/dockertest@v3.3.5+incompatible/docker/client.go:371 +0x192
  github.com/ory/dockertest/docker.(*Client).startContainer()
      /home/runner/go/pkg/mod/github.com/ory/dockertest@v3.3.5+incompatible/docker/container.go:828 +0xc7
  github.com/ory/dockertest/docker.(*Client).StartContainer()
      /home/runner/go/pkg/mod/github.com/ory/dockertest@v3.3.5+incompatible/docker/container.go:808 +0x1087
  github.com/ory/dockertest.(*Pool).RunWithOptions()
      /home/runner/go/pkg/mod/github.com/ory/dockertest@v3.3.5+incompatible/dockertest.go:320 +0x1021
  ...
  testing.tRunner()
      /opt/hostedtoolcache/go/1.17.0/x64/src/testing/testing.go:1259 +0x22f
  testing.(*T).Run·dwrap·21()
      /opt/hostedtoolcache/go/1.17.0/x64/src/testing/testing.go:1306 +0x47

Previous write at 0x00c0001622b8 by goroutine 44:
  github.com/ory/dockertest/docker.(*Client).checkAPIVersion()
      /home/runner/go/pkg/mod/github.com/ory/dockertest@v3.3.5+incompatible/docker/client.go:371 +0x192
  github.com/ory/dockertest/docker.(*Client).startContainer()
      /home/runner/go/pkg/mod/github.com/ory/dockertest@v3.3.5+incompatible/docker/container.go:828 +0xc7
  github.com/ory/dockertest/docker.(*Client).StartContainer()
      /home/runner/go/pkg/mod/github.com/ory/dockertest@v3.3.5+incompatible/docker/container.go:808 +0x1087
  github.com/ory/dockertest.(*Pool).RunWithOptions()
      /home/runner/go/pkg/mod/github.com/ory/dockertest@v3.3.5+incompatible/dockertest.go:320 +0x1021
  ...
  testing.tRunner()
      /opt/hostedtoolcache/go/1.17.0/x64/src/testing/testing.go:1259 +0x22f
  testing.(*T).Run·dwrap·21()
      /opt/hostedtoolcache/go/1.17.0/x64/src/testing/testing.go:1306 +0x47
```

**To Reproduce**

1. Write multiple tests and call `pool.RunWithOptions` make sure to run tests in parallel `t.Parallel()`
2. Run test with race detector `go test -race`

**Expected behavior**

I expected there to be no possibility for race conditions originating in the dockertest package when starting multiple containers concurrently.

**Environment**

- Version: v3.3.5
- Environment: MacOS Big Sur, Docker Engine 20.10.7

**Additional context**

The race condition is triggered from the below block of code. The side effect of setting `Client.serverAPIVersion` in `Client.checkAPIVersion` is not concurrency safe.

https://github.com/ory/dockertest/blob/92472ba5fc3e0e00f77069df607dd69abee5c36d/docker/container.go#L827-L833

**Comment by aeneasr** [comment]
Nice find :)

**Comment by ahmedalhulaibi** [comment]
Happy to help. I recreated the issue in this commit https://github.com/ory/dockertest/pull/292/commits/8384f045142f2948b511b49697fc40a27dbd19db

It is fixed in the latest commit of my PR #292
